core/src/main/java/software/amazon/awssdk/core/protocol/json/JsonContentTypeResolver.java

```diff
@@ -21,27 +21,7 @@
  * Interface to compute the content type to send in requests for JSON based protocols.
  */
 @SdkInternalApi
-interface JsonContentTypeResolver {
-
-    /**
-     * Content type resolver implementation for Ion-enabled services.
-     */
-    JsonContentTypeResolver ION_BINARY = new JsonContentTypeResolverImpl("application/x-amz-ion-");
```

Comment:
For stuff like this I don't necessarily care where it lives. It can be in core if that means we can hide the impl from module boundaries. Don't care either way though.